plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -188,9 +216,14 @@ public TableInfo getCachedTable(Duration viewExpiration, TableInfo remoteTableId
         return materializationCache.getCachedTable(this, query, viewExpiration, remoteTableId);
     }
 
+    public String getParentProjectId()
+    {
+        return this.parentProjectId;
+    }
+
     public String getProjectId()
     {
-        String projectId = configProjectId.orElseGet(() -> bigQuery.getOptions().getProjectId());
+        String projectId = configProjectId.orElseGet(this::getParentProjectId);
```

Comment:
Is this change required to fix the issue you faced? When does `bigQuery.getOptions().getProjectId()` return a different value from  `setProjectId(parentProjectId)` at L109?

Reply:
It's not required to fix the issue, but I think it would bring meaning to what `bigQuery.getOptions().getProjectId()` inside that `orElseGet` really is and avoids dup.